src/main/java/com/pusher/rest/PusherAbstract.java

```diff
@@ -272,16 +324,22 @@ public T trigger(final List<String> channels, final String eventName, final Obje
      */
     public T trigger(final List<Event> batch) {
         final List<Event> eventsWithSerialisedBodies = new ArrayList<Event>(batch.size());
+
         for (final Event e : batch) {
+            final String eventData = shouldEncryptPayloads(e.getChannel())
+                ? encryptMessage(e.getChannel(), serialise(e.getData()))
+                : serialise(e.getData());
```

Comment:
`shouldEncryptPayloads` bad choice of method name perhaps! otherwise it checks channel name and also the presence of the master key.

Reply:
I think the name is fine, but the other `trigger` method throws an exception if the channel name indicates that encryption is required and there is no valid master key. It feels like we should be consistent between the two methods, and my instinct is that an exception is the right way to go (otherwise people might believe their channels are encrypted when they are not).